### PR TITLE
Fix for JENKINS-24581

### PR DESCRIPTION
**Pull request by pbthorste**
These changes update the batch-task plugin so that it displays correctly in more recent versions of jenkins.

### CODE REVIEW

File: src/main/java/hudson/plugins/batch_task/BatchTask.java [hunk 1]
```diff
@@ -378,4 +381,10 @@ public Task getOwnerTask() {
     public Object getSameNodeConstraint() {
         return null;
     }
+    
+    @Override
+    public Authentication getDefaultAuthentication() {
```
Comment (daniel-beck) [inline src/main/java/hudson/plugins/batch_task/BatchTask.java:386]: What is this?

Comment (pbthorste) [inline src/main/java/hudson/plugins/batch_task/BatchTask.java:386]: Without it I get the error: 

> The type BatchTask must implement the inherited abstract method Queue.Task.getDefaultAuthentication()

**Comment by pingbat** [comment]
Hello, I manually applied the patches from this PR and tried it out. The display is much better but starting tasks stopped working. My task seemed to be waiting for an executor to become available but nothing else was running.

**Comment by pbthorste** [comment]
What version of Jenkins did you run the plugin on ? I have tested it on 1.615 and so far it has been working.

**Comment by pingbat** [comment]
I'm on the latest 1.617:

image

image

I may have made a mistake in applying the patches (I couldn't see an easy way to grab a ZIP file of the PR, I can't install GitHub for Windows)

I'll see if I can get the code onto a different machine and try it there.

**Comment by pingbat** [comment]
I've just checked my logs and I see this:

Jun 26, 2015 9:12:42 AM hudson.triggers.SafeTimerTask run
SEVERE: Timer task hudson.model.Queue$MaintainTask@cc1957 failed
java.lang.IllegalArgumentException: Authentication required
        at org.springframework.util.Assert.notNull(Assert.java:112)
        at org.acegisecurity.acls.sid.PrincipalSid.<init>(PrincipalSid.java:45)
        at hudson.security.SidACL._hasPermission(SidACL.java:70)
        at hudson.security.SidACL.hasPermission(SidACL.java:52)
        at hudson.model.Node.canTake(Node.java:366)
        at hudson.model.Queue$JobOffer.canTake(Queue.java:296)
        at hudson.model.Queue.maintain(Queue.java:1389)
        at hudson.model.Queue$MaintainTask.doRun(Queue.java:2489)
        at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:51)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask$Sync.innerRunAndReset(FutureTask.java:351)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
        at java.lang.Thread.run(Thread.java:722)

**Comment by pingbat** [comment]
I've changed BatchTask.getDefaultAuthentication() to this and it works for me now:

@Override
 public Authentication getDefaultAuthentication() {
      return ACL.SYSTEM;
}

**Comment by pingbat** [comment]
Any thoughts on merging this in with the ACL.SYSTEM change? I've been using it fine for months now.

**Comment by pbthorste** [comment]
Sorry for taking so long - but the change is pushed

File: src/main/java/hudson/plugins/batch_task/BatchTask.java (hunk 1 above)
Comment (pbthorste) [inline src/main/java/hudson/plugins/batch_task/BatchTask.java:386]: Any chance you could take another look, daniel-beck ?

**Comment by michaelpaesold** [comment]
We have been using a 1.18 snapshot build of the plugin based on the latest commit 3b0f9f49fa1a87f626945bc05da7b31a965c899d in production successfully since end of 2015, running on Jenkins LTS versions 1.609.3 and later 1.625.3. We will upgrade to newer Jenkins versions soon.

I think this PR is good and should be merged. Are there any reasons not to do so, other than resolving conflicts?

The released version 1.19 does **not** display correctly on the Jenkins versions above.

**Comment by michaelpaesold** [comment]
@pbthorste - would you update the PR? Otherwise I could take your branch and do that myself.

**Comment by michaelpaesold** [comment]
I have created a new rebased PR #7 that only contains the actually required changes compared to current master.

**Comment by lacostej** [comment]
Closing this as the code was merged in #12. Thanks @pbthorste

**Pull request closed without merging**